src/Xamarin.Android.Build.Tasks/Tasks/Javac.cs

```diff
@@ -57,16 +58,19 @@ protected override string GenerateCommandLineCommands ()
 
 			cmd.AppendSwitchIfNotNull ("-J-Dfile.encoding=", "UTF8");
 
-			cmd.AppendSwitchIfNotNull ("-d ", ClassesOutputDirectory);
-
-			cmd.AppendSwitchIfNotNull ("-classpath ", Jars == null || !Jars.Any () ? null : string.Join (Path.PathSeparator.ToString (), Jars.Select (i => i.ItemSpec)));
-			cmd.AppendSwitchIfNotNull ("-bootclasspath ", JavaPlatformJarPath);
-			cmd.AppendSwitchIfNotNull ("-encoding ", "UTF-8");
 			cmd.AppendFileNameIfNotNull (string.Format ("@{0}", TemporarySourceListFile));
 			cmd.AppendSwitchIfNotNull ("-target ", JavacTargetVersion);
 			cmd.AppendSwitchIfNotNull ("-source ", JavacSourceVersion);
 
 			return cmd.ToString ();
 		}
+
+		protected override void WriteOptionsToResponseFile (StreamWriter sw)
+		{
+			sw.WriteLine ($"-d {Files.GetShortPath (ClassesOutputDirectory)}");
+			sw.WriteLine ("-classpath \"{0}\"", Jars == null || !Jars.Any () ? null : string.Join (Path.PathSeparator.ToString (), Jars.Select (i => i.ItemSpec)));
+			sw.WriteLine ("-bootclasspath {0}", Files.GetShortPath (JavaPlatformJarPath));
```

Comment:
I'm 99% sure that this will fail on Windows, because of my old favorite:

https://github.com/xamarin/xamarin-android/blob/38bc7d0d7705b580c8e0941affb3954b7de7d602/src/Xamarin.Android.Build.Tasks/Tasks/JavaCompileToolTask.cs#L77-L93

`javac` *does not like* `\` in the response file: it treats it as an *escape* character (?!).  We fix this in `<JavaCompileToolTask/>` by replacing `\` with `\\`, thus appeasing `javac` on Windows, but *this* line won't go through that fixup logic, meaning this *should* generate:

```
-bootclasspath C:\some\path\to\android.jar
```

which will *not* work as desired.